### PR TITLE
Context api tests

### PR DESCRIPTION
**Pull request by ptitSeb**
# Description
Migrate compiler tests and metering to the new Context API

### CODE REVIEW

File: lib/middlewares/src/metering.rs [hunk 1]
```diff
@@ -418,21 +411,21 @@ mod tests {
...
         add_one.call(&mut ctx.as_context_mut(), 1).unwrap();
         assert_eq!(
-            get_remaining_points(ctx.as_context_mut(), &instance),
+            get_remaining_points(&mut ctx.as_context_mut(), &instance),
             MeteringPoints::Remaining(2)
         );
 
         // Third call fails due to limit
-        assert!(add_one.call(&mut ctx.as_context_mut(), 1).is_err());
+        assert!(add_one.call(1).is_err());
```
Comment (syrusakbary) [inline lib/middlewares/src/metering.rs:426]: Is this call valid?

Comment (ptitSeb) [reply lib/middlewares/src/metering.rs]: No. That's strange. Something must have gone wrong when I  rebased the branch over context-api.

**Review by epilys — CHANGES_REQUESTED** [review]
Squash fixes with previous commits

**Review by syrusakbary — CHANGES_REQUESTED** [review]
Let's change all the calls to `&mut ctx.as_context_mut()` to `&mut ctx` (and from `&ctx.as_context_ref()` to `&ctx`, if any of this exists)

**Comment by ptitSeb** [comment]
> Squash fixes with previous commits

Done

**Comment by ptitSeb** [comment]
> Let's change all the calls to `&mut ctx.as_context_mut()` to `&mut ctx` (and from `&ctx.as_context_ref()` to `&ctx`, if any of this exists)

Done

**Review by syrusakbary — APPROVED** [review]

**Pull request merged**